tests/func/test_ignore.py

```diff
@@ -403,8 +404,25 @@ def test_ignore_in_added_dir(tmp_dir, dvc):
     assert not ignored_path.exists()
 
 
-def test_ignored_output(tmp_dir, scm, dvc, run_copy):
-    tmp_dir.gen({".dvcignore": "*.log", "foo": "foo content"})
+@pytest.mark.parametrize(
+    "ignore_patterns,raise_error",
+    [("*.log", True), ("*.log\n!foo.log", False)],
+)
+def test_ignored_output(
+    tmp_dir, scm, dvc, run_copy, ignore_patterns, raise_error
+):
+    tmp_dir.gen({".dvcignore": ignore_patterns, "foo": "foo content"})
 
-    with pytest.raises(OutputIsIgnoredError):
+    if raise_error:
+        with pytest.raises(OutputIsIgnoredError):
+            run_copy("foo", "foo.log", name="copy")
+    else:
```

Comment:
Not sure we need to parametrize this test if we need `if` condition. We can split it into two tests, or go with
`try/catch` block, and verifying type of exception thrown, that way we don't have to put condition depending on parameters.